hook.js

```diff
@@ -9,68 +10,75 @@ var parseSource = require('./sourceParse');
+       printDebug(metadata);
+       comment(metadata);
     }
+}
 
-    // TODO: Only acknowledge pushes to the "Master" branch.
-    console.log('Repo: ' + payload.repository.full_name);
-    console.log('Payload:');
-    console.log(payload);
+function comment(metadata) {
+    var pullRequestNumber = metadata.number;
+    var destinationRepository = metadata.repository.full_name;
+    var originRepository = metadata.pull_request.head.repo.full_name;
+    var originBranch = metadata.pull_request.head.ref;
 
-    sendRequest({
-        url: configMetadataURL,
-        headers: {
-            'User-Agent': productName
-        }
-    }, function(error, response, body) {
-        var configMetadata = JSON.parse(body);
-        var config = ['last 2 versions'];
-        var commitsURL = payload.pull_request.commits_url;
-        var commentURL = payload.pull_request.review_comments_url;
-        var configMetadataContent;
-
-        if (configMetadata.content) {
-            configMetadataContent = new Buffer(configMetadata.content, 'base64').toString();
-
-            // Consider text separated by commas and linebreaks to be individual
-            // options
-            config = configMetadataContent.replace(/\r?\n|\r/g, ', ').split(/,\s*/);
-        }
+    trackUsage(originRepository);
+
+    getConfiguration(originRepository, originBranch, function(config) {
+        var prClient = githubClient.pr(destinationRepository, pullRequestNumber);
+
+        prClient.commits(function(error, commits) {
+            if(error) return console.error(error);
 
-        sendRequest({
-            url: commitsURL,
-            headers: {
-                'User-Agent': productName
-            }
-        }, function(error, response, body) {
-            var commits = JSON.parse(body);
-
-            commits.forEach(function(element, index) {
-                sendRequest({
-                    url: element.url,
-                    headers: {
-                        'User-Agent': productName
-                    }
-                }, function(error, response, body) {
-                    var commitMeta = JSON.parse(body);
-                    var commitFiles = commitMeta.files;
-                    var commitSHA = commitMeta.sha;
-                    parseCSS(commitFiles, config, commentURL, token, function(usageInfo) {}, commitSHA);
+            commits.forEach(function(currentCommit) {
+                var originRepoClient = githubClient.repo(originRepository);
+                originRepoClient.commit(currentCommit.sha, function(error, currentCommitDetail) {
+                    if(error) return console.error(error);
+                    parseCSS(currentCommitDetail.files, config, metadata.pull_request.review_comments_url, token, function(usageInfo) {}, currentCommit.sha);
                 });
             });
         });
     });
-};
+}
+
+function getConfiguration(repository, branch, callback) {
+    var repositoryClient = githubClient.repo(repository);
```

Comment:
Can we short to "repoClient"?  The var name with the spelled out version is a bit verbose IMO